nav2_rviz_plugins/src/goal_tool.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_rviz_plugins/goal_tool.hpp"
+
+#include <memory>
+#include <string>
+
+#include "rviz_common/display_context.hpp"
+#include "rviz_common/load_resource.hpp"
+
+namespace nav2_rviz_plugins
+{
+
+GoalTool::GoalTool()
+: rviz_default_plugins::tools::PoseTool()
+{
+  shortcut_key_ = 'g';
+
+  navigation_dialog_ = std::make_unique<NavigationDialog>();
+  navigation_dialog_->move(0, 0);
+}
+
+GoalTool::~GoalTool()
+{
+}
+
+void GoalTool::onInitialize()
+{
+  PoseTool::onInitialize();
+  setName("2D Nav2 Goal");
```

Comment:
```suggestion
  setName("Navigation2 Goal");
```
Minor suggestion just to make the button different enough from the other (existing) button "2D Nav Goal". 

Also, as an aside, what happens if the other one is pressed? Do we still have support for the 'move_base_simple/goal' topic?

Reply:
Text changed to "Navigation2 Goal"

Yes, the old topic-based goal tool is still supported. The new goal tool uses actions and supports canceling.